torch/csrc/autograd/variable.h

```diff
@@ -757,53 +669,17 @@ inline bool Variable::is_view() const noexcept {
 inline const Variable& Variable::base() const {
   if (is_view()) {
     // is_view() implies get_autograd_meta()
-    auto diff_view_meta = static_cast<DifferentiableViewMeta*>(get_autograd_meta());
+    auto diff_view_meta = static_cast<DifferentiableViewMeta*>(impl::get_autograd_meta(*this));
     return diff_view_meta->base_;
   } else {
     throw std::runtime_error("Can't get base of non-view Variable");
   }
 }
 
-// Miscellaneous
-//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-inline void Variable::set_name(const std::string& name) {
-  materialize_autograd_meta()->name_ = name;
-}
-
-namespace {
-  std::string singleton_string;
-}
-
-inline const std::string& Variable::name() const noexcept {
-  if (get_autograd_meta()) {
-    return get_autograd_meta()->name_;
-  } else {
-    return singleton_string;
-  }
-}
-
-inline void Variable::set_pyobj(PyObject* pyobj) noexcept {
-  get()->set_pyobj(pyobj);
-}
-
-inline PyObject* Variable::pyobj() const noexcept {
-  return get()->pyobj();
-}
-
-inline AutogradMeta* Variable::get_autograd_meta() const noexcept {
-  // NB: could return null
-  return static_cast<AutogradMeta*>(get()->autograd_meta());
-}
-
 // Private Methods
 //~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 inline Variable::Variable(c10::intrusive_ptr<at::TensorImpl> self)
     : at::Tensor(std::move(self)) {}
 
-inline at::TensorImpl* Variable::get() const {
-  TORCH_CHECK(defined(), "Called Variable::get() on an undefined Variable");
```

Comment:
It's probably safer to add definedness checks to all of the sites that used to call get()

Reply:
Actually, all of the implementations that get moved to VariableType are safe because if you have an undefined tensor you won't dispatch here in the first place.

Reply:
For the other ones, I stacked it as a diff on top as incorporating into this diff would have resulted in some merge conflicts. https://github.com/pytorch/pytorch/pull/29907